p2p_utils.py

```diff
@@ -44,3 +46,22 @@ async def check_reachability_parallel(peer_ids, dht, node, *, fetch_info=False):
         *[check_reachability(peer_id, dht, node, fetch_info=fetch_info) for peer_id in peer_ids]
     )
     return dict(zip(peer_ids, rpc_infos))
+
+
+async def get_peers_ips(dht, dht_node):
+    return await dht_node.p2p.list_peers()
+
+@functools.cache
+def get_location(ip_address):
+    try:
+        response = requests.get(f"http://ip-api.com/json/{ip_address}")
+        if response.status_code == 200:
+            return response.json()
+    except Exception:
+        pass
+    return {}
+
+def extract_peer_ip_info(multiaddr_str):
+    if ip_match := re.search(r"/ip4/(\d+\.\d+\.\d+\.\d+)", multiaddr_str):
+        return get_location(ip_match[1])
+    return {}
```

Comment:
Not sure if this is a necessary feature for the health monitor itself: I think we can return multiaddrs of each node via the API, and then the requesting code can do anything it needs with them. That way, we will not need to have application-specific details in the API code

Reply:
I agree with you, but if we do that frontend wise, it will take more time to load the first time and the experience could be kind of slow.

Reply:
I am caching all the responses. No slow down. You can test it locally. I mostly agree with you, but this PR is not creating any conflict and it's adding information that could be valuable for other developers too.

Reply:
Your alternative solution would mean that we need to run a webserver on top of a frontend application in order to add the metadata. I think it's an over complicated solution for a super simple use case.